core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1646,7 +1646,7 @@ class SparkContext(config: SparkConf) extends Logging {
   def killExecutors(executorIds: Seq[String]): Boolean = {
     schedulerBackend match {
       case b: CoarseGrainedSchedulerBackend =>
-        b.killExecutors(executorIds, replace = false, force = true).nonEmpty
+        b.killExecutors(executorIds, replace = false, force = false).nonEmpty
```

Comment:
I don't think this should change. This is the user's code explicitly asking for executors to be killed, so there's no reason to not honor that request. It would also change the behavior of a public API, and there's no way for the user to know which executors were killed or not from the method's return value.